From 4b25008ec5afc550231c4b0032bd9a581d51ee36 Mon Sep 17 00:00:00 2001
From: Henri Yandell <bayard@apache.org>
Date: Tue, 19 Sep 2006 08:18:02 +0000
Subject: [PATCH] #LANG-59 fails under JDK 1.3, so much as with the earlier
 code in this class, I've modified it so that it only runs with 1.4 and above.
 A warning is output, and I improved the other warning message at the same
 time.

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/lang/trunk@447778 13f79535-47bb-0310-9956-ffa450edef68
---
 src/test/org/apache/commons/lang/time/DateUtilsTest.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/test/org/apache/commons/lang/time/DateUtilsTest.java b/src/test/org/apache/commons/lang/time/DateUtilsTest.java
index 7f9b5a417c8..6f6a434c51c 100644
--- a/src/test/org/apache/commons/lang/time/DateUtilsTest.java
+++ b/src/test/org/apache/commons/lang/time/DateUtilsTest.java
@@ -686,7 +686,7 @@ public void testRound() throws Exception {
                     dateTimeParser.parse("March 30, 2003 04:00:00.000"),
                     DateUtils.round((Object) cal7, Calendar.HOUR_OF_DAY));
         } else {
-            this.warn("Some date rounding tests not run since the current version is " + SystemUtils.JAVA_VERSION);
+            this.warn("WARNING: Some date rounding tests not run since the current version is " + SystemUtils.JAVA_VERSION);
         }
         TimeZone.setDefault(defaultZone);
         dateTimeParser.setTimeZone(defaultZone);
@@ -889,6 +889,10 @@ public void testTruncate() throws Exception {
      * see http://issues.apache.org/jira/browse/LANG-59
      */
     public void testTruncateLang59() throws Exception {
+        if (!SystemUtils.isJavaVersionAtLeast(1.4f)) {
+            this.warn("WARNING: Test for LANG-59 not run since the current version is " + SystemUtils.JAVA_VERSION);
+            return;
+        }
 
         // Set TimeZone to Mountain Time
         TimeZone MST_MDT = TimeZone.getTimeZone("MST7MDT");
